test/runnable/test11934.d

```diff
@@ -0,0 +1,22 @@
+void main()
+{
+    struct Struct11934
+    {
+        this(int i) { instances++; }
+        this(this) { instances++; }
+        ~this() { instances--; }
+        static size_t instances = 0;	
+    }
+
+    struct Range11934
+    {
+        void popFront() { cnt++; }
+        @property front() { return Struct11934(0); }
+        @property empty() { return cnt >= 10; }
+        size_t cnt;
+    }
+
+    foreach(ref i; Range11934()) { }
+
+	assert(Struct11934.instances == 0);
```

Comment:
Maybe have three global counters to ensure the {con,de,copy}structors are actually called.

Reply:
Imo it's not the point of the test for this issue, but I can do it easily of course.

Reply:
In this example postblit is never called but counter is there to make clear we're not missing any check.